test/parse.js

```diff
@@ -366,6 +387,42 @@ test('parse', function (t) {
             { type: 'text', content: 'There' }
         ]
     }], 'should remove text nodes that are nothing but whitespace');
+
+    html = '<div>Hi</span>There</div>';
+    parsed = HTML.parse(html);
+    t.deepEqual(parsed, [{
+        type: 'tag',
+        name: 'div',
+        attrs: {},
+        voidElement: false,
+        children: [
+            { type: 'text', content: 'Hi' },
+            { type: 'text', content: 'There' }
+        ]
+    }], 'should skip over closing tags that don\'t match the current tag name');
```

Comment:
So, I tried throwing this html into a file and seeing how Chrome and Firefox parsed it. They both ignored the </span>, and formed a single text node. Can we have the same behavior as those browser in the parser?

<img width="194" alt="Screen Shot 2020-01-30 at 2 48 43 PM" src="https://user-images.githubusercontent.com/469808/73497440-a95b8680-436f-11ea-8471-9a97f827daf3.png">